Updated registration/log in

mayasa3 opened the pull request:
- Updated views and HTML pages
- Added CSS to log in, registration, and account pages

Fixes #86
Resolves #87
Resolves #88

@bardielle @guy9050

guy9050 commented on accounts/views.py:
```diff
@@ -14,10 +27,10 @@ def sign_up_view(request):
 
 @login_required
 def account_view(request):
+    """ Account UI - work in progress """
```
what still in progress?

guy9050 commented on accounts/views.py:
```diff
@@ -14,10 +27,10 @@ def sign_up_view(request):
 
 @login_required
 def account_view(request):
+    """ Account UI - work in progress """
```
if it's in progress why the PR isn't [WIP]?

mayasa3 replied:
I added the page so that the UI can work, but it doesn't have content yet(that will be added later, probably in a separate PR once some more future PRs are merged from everyone). Should I remove the work in progress line or mark this as work in progress until all the rest of the features are completed?

bardielle replied:
You should mention the specific issue for that...

mayasa3 replied:
Should I update the comment in the file with the issue? Or in the PR/commit description? I only added comments here, but haven't done anything with this view in any of these commits

bardielle replied:
In the file

mayasa3 replied:
Done

itaiv95 commented on accounts/templates/account.html:
```diff
@@ -2,41 +2,30 @@
-		</p>
-		
-		<p>
-		<a href="{% url 'logout' %}"> Log Out </a>
```
why we need this page?
wht not dierct the user to the search page after login sucssusfuly?

bardielle commented:
@mayasa3 let us know when 2 (at least) of your team mates reviewed your code

mayasa3 commented:
> @mayasa3 let us know when 2 (at least) of your team mates reviewed your code
Done

bardielle commented on accounts/views.py:
```diff
@@ -4,7 +4,20 @@
 
+def login_excluded(redirect_to):
+    """ Kicks authenticated users out of a view """
+    def _method_wrapper(view_method):
```
why do you need function in function in function? 
I think you need to create a class here instead

mayasa3 replied:
When I looked up how to do it the results always had functions in functions, didn't find much about making it a class and so far I haven't managed to get it to work as a class. Should I remove it and try to find another way to do this?

bardielle replied:
I think it's not clear like that 
@guy9050 what do you think?

guy9050 replied:
agree, @mayasa3 I see that you deleted it.. do you not need it eventually?

mayasa3 replied:
I didn't delete it, but I moved it to a different file so that it(and other custom decorators if anyone needs/wants to add them) won't be limited to one app. 
Right now I mostly just need to keep logged in users out of these pages, I can try a different way but I thought making it similar to @login_required was a better idea at the time

bardielle replied:
So please take a look in other projects
you can't keep it like that

mayasa3 replied:
Deleted it now

bardielle replied:
Does it work?

bardielle replied:
@mayasa3 ?

mayasa3 replied:
It works on my end

mayasa3 replied:
Works on my end but I'll double check to make sure
EDIT: Everything works so far @bardielle

mayasa3 replied:
I changed sign up to a class-based view while I was at it, everything still works so far

mayasa3 replied:
@bardielle @guy9050 does it look ok now?

bardielle replied:
Yes

guy9050 commented:
lgtm

mayasa3 committed:
CSS file for accounts app

mayasa3 committed:
Updated HTML pages

Added CSS and updated code

mayasa3 committed:
Updated views

- Sign up and log in changed to class based views

- Updated code for keeping logged in users out

bardielle reviewed (approved):
/lgtm
